dffml/df/base.py

```diff
@@ -216,8 +216,10 @@ def wrap(func):
                         ),
                     )
                 else:
```

Comment:
This should be in an `elif` where we check if the definition is a NamedTuple, or if it's a dataclass, if it is either, then we do this new code you've added. Otherwise we keep the old else statement.

be careful though because you can't just do `issubclass(..., NamedTuple)`,

```python
>>> from typing import NamedTuple
>>> import dataclasses
>>>
>>> class SubN(NamedTuple):
...   i: int
...
>>> @dataclasses.dataclass
... class SubD:
...   i: int
...
>>> print(bool(issubclass(SubN, NamedTuple)))
False
>>> print(dataclasses.is_dataclass(SubD))
True
>>> print(bool(issubclass(SubN, tuple) and hasattr(SubN, "_asdict")))
True
>>>
```

Reply:
done, thanks